src/DeviceDataStreamValues.tsx

```diff
@@ -0,0 +1,219 @@
+  }
+  if (_.isBoolean(data)) {
+    return data ? 'true' : 'false';
+  }
+  if (_.isNumber(data)) {
+    return data.toString();
+  }
+  if (_.isNull(data)) {
+    return '';
+  }
+  return String(data);
+}
+
+interface IndividualDatastreamTableProps {
+  data: AstarteDatastreamIndividualData[];
+}
+
+const IndividualDatastreamTable = ({
+  data,
+}: IndividualDatastreamTableProps): React.ReactElement => {
+  return (
+    <Table responsive>
+      <thead>
+        <tr>
+          <th>Path</th>
+          <th>Value</th>
+          <th>Timestamp</th>
+        </tr>
+      </thead>
+      <tbody>
+        {data.map((tree, index) => (
+          <tr key={index}>
+            <td>{tree.endpoint}</td>
+            <td>{tree.value}</td>
+            <td>{new Date(tree.timestamp).toLocaleString()}</td>
+          </tr>
+        ))}
+      </tbody>
+    </Table>
+  );
+};
+
+interface ObjectDatastreamTableProps {
+  data: AstarteDatastreamObjectData[];
+}
+
+const ObjectDatastreamTable = ({ data }: ObjectDatastreamTableProps): React.ReactElement => {
+  const objectProperties = Object.keys(data[0].value);
+  return (
+    <>
+      <Table responsive>
+        <thead>
+          <tr>
+            {objectProperties.map((property) => (
+              <th key={property}>{property}</th>
+            ))}
+            <th>Timestamp</th>
+          </tr>
+        </thead>
+        <tbody>
+          {data.map((tree) => (
+            <tr key={tree.timestamp}>
+              {objectProperties.map((property) => (
+                <td key={property}>{formatAstarteData(tree.value[property])}</td>
+              ))}
+              <td>{new Date(tree.timestamp).toLocaleString()}</td>
+            </tr>
+          ))}
+        </tbody>
+      </Table>
+    </>
+  );
+};
+
+interface InterfaceDataProps {
+  interfaceData: AstarteDatastreamIndividualData[] | AstarteDatastreamObjectData[];
+  aggregation: AstarteInterface['aggregation'];
+}
+
+const InterfaceData = ({ interfaceData, aggregation }: InterfaceDataProps): React.ReactElement => {
+  if (interfaceData.length === 0) {
+    return <p>No data in the selected timeframe.</p>;
+  } else if (aggregation === 'individual') {
+    return <IndividualDatastreamTable data={interfaceData as AstarteDatastreamIndividualData[]} />;
+  }
+  return <ObjectDatastreamTable data={interfaceData as AstarteDatastreamObjectData[]} />;
+};
+
+export default (): React.ReactElement => {
+  const { interfaceName = '', deviceId = '', interfaceMajor } = useParams();
+  const astarte = useAstarte();
+  const [requestingData, setRequestingData] = useState(false);
+  const [fetchedData, setFetchedData] = useState<AstarteInterfaceValues | null>(null);
+  const [formAlerts, formAlertsController] = useAlerts();
+  const [interfaceData, setInterfaceData] = useState<AstarteInterface | null>(null);
+
+  const deviceDataFetcher = useFetch(() =>
+    astarte.client.getDeviceData({
+      deviceId,
+      interfaceName,
+    }),
+  );
```

Comment:
Why are you fetching all interface data here? Is it just to know which endpoints exist?
In that case, I suggest computing the existing endpoints as soon as possible to use them in other components, and avoid using `deviceDataFetcher` in many places.